public/components/trace_analytics/components/traces/span_detail_table.tsx

```diff
@@ -253,32 +299,42 @@ export function SpanDetailTable(props: SpanDetailTableProps) {
 }
 
 export function SpanDetailTableHierarchy(props: SpanDetailTableProps) {
-  const { http, hiddenColumns, mode, DSL, dataSourceMDSId, availableWidth, openFlyout } = props;
+  const { hiddenColumns, mode, availableWidth, openFlyout } = props;
   const [items, setItems] = useState<Span[]>([]);
   const [_total, setTotal] = useState(0);
   const [expandedRows, setExpandedRows] = useState(new Set<string>());
   const [isSpansTableDataLoading, setIsSpansTableDataLoading] = useState(false);
 
   useEffect(() => {
-    setIsSpansTableDataLoading(true);
-    const spanSearchParams = {
-      from: 0,
-      size: 10000,
-      sortingColumns: [],
-    };
-    handleSpansRequest(
-      http,
-      (data) => {
-        const hierarchy = buildHierarchy(data);
-        setItems(hierarchy);
-      },
-      setTotal,
-      spanSearchParams,
-      DSL,
-      mode,
-      dataSourceMDSId
-    ).finally(() => setIsSpansTableDataLoading(false));
-  }, [DSL, http, mode, dataSourceMDSId]);
+    if (!props.payloadData) return;
+    try {
+      const parsed = JSON.parse(props.payloadData);
+      const hitsArray =
+        parsed.hits && Array.isArray(parsed.hits.hits)
+          ? parsed.hits.hits
+          : Array.isArray(parsed)
+          ? parsed
+          : [];
```

Comment:
same here

Reply:
is there a reason that this is different from https://github.com/opensearch-project/dashboards-observability/blob/0fe071019eef38123533982bb9183a0b874d4521/public/components/trace_analytics/components/traces/span_detail_panel.tsx#L144-L149? and maybe extract the parsing into a util function?